tools/post-potd/post-potd.py

```diff
@@ -0,0 +1,37 @@
+from discord import SyncWebhook
+
+import json
+import requests
+
+from bs4 import BeautifulSoup
+
+
+def get_last_potd():
+    response = requests.get("https://leetcode.com/problemset/all/")
+    if response.status_code != 200:
+        raise "oops"
+    soup = BeautifulSoup(response.text, features="html.parser")
+    data = json.loads(soup.find(id="__NEXT_DATA__").text)
+    queries = data["props"]["pageProps"]["dehydratedState"]["queries"]
+    relevant_queries_data = [
+        item
+        for q in queries
+        if q["queryKey"][0] == "dailyCodingQuestionRecords"
+        for item in q["state"]["data"]["dailyCodingChallengeV2"]["challenges"]
+    ]
+    return max(relevant_queries_data, key=lambda item: item["date"])
+
+
+def main():
```

Comment:
Remoe the function and put it into the block on line 36.

Reply:
What do you suggest instead?

Reply:
I was thinking put the code into the " if __name__ == "__main__": " statement, but it would probably be better to just give the function a better name like "post_potd"

Reply:
Hmm, I can see multiple arguments. On the one hand calling it `main` seems idiomatic according to https://realpython.com/python-main-function/
On the other, it would work just as well with a more specific name. What about keeping the name `main` but adding a comment about what it does?